src/globus_cli/commands/search/index/role/create.py

```diff
@@ -0,0 +1,85 @@
+import uuid
+from typing import Optional
+
+import click
+
+from globus_cli.login_manager import LoginManager
+from globus_cli.parsing import command
+from globus_cli.termio import FORMAT_TEXT_RECORD, formatted_print
+
+from ..._common import index_id_arg, resolved_principals_field
+
+
+@command("create")
+@index_id_arg
+@click.argument("ROLE_NAME")
+@click.argument("PRINCIPAL")
+@click.option(
+    "--type",
+    "principal_type",
+    type=click.Choice(("identity", "group")),
+    help=(
+        "The type of the principal. "
+        "If the principal is given as a URN, it will be checked against any provided "
+        "'type'. If a non-URN string is given, the type will be used to format the "
+        "principal as a URN."
+    ),
+)
+@LoginManager.requires_login(LoginManager.SEARCH_RS, LoginManager.AUTH_RS)
+def create_command(
+    *,
+    index_id: uuid.UUID,
+    role_name: str,
+    principal: str,
+    principal_type: Optional[str],
+    login_manager: LoginManager,
+):
+    """
+    Create a role (requires admin or owner)
+
+    PRINCIPAL is expected to be an identity or group ID, a principal URN, or a username.
+
+    Example usage:
+       globus-search index role create admin "$index_id" "globus@globus.org"
+       globus-search index role create writer "$index_id" "$group_id" --type group
```

Comment:
I think the arguments need to have "$index_id" before the ROLE_NAME, unless this is the intended order and the argument decorators are swapped?

Reply:
No, your first guess correct. It was meant to be `"$index_id" admin`, and likewise for `writer`.
Thanks for the catch; I'll fix this.